src/include/storage/tuple_access_strategy.h

```diff
@@ -253,6 +253,40 @@ class TupleAccessStrategy {
    */
   const BlockLayout &GetBlockLayout() const { return layout_; }
 
+  /**
+   * Compare and swap block status to busy. Expect that block to be idle, if yes, set the block status to be busy
+   * @param block the block to compare and set
+   * @return true if the set operation succeeded, false if the block is already busy
+   */
+  bool SetBlockBusyStatus(RawBlock *block) const {
+    uint32_t old_val = clr_bit(block->insert_head_.load());
+    return block->insert_head_.compare_exchange_weak(old_val, set_bit(old_val));
+  }
+
+  /**
+   * Compare and swap block status to idle. Expect that block to be busy, if yes, set the block status to be idle
+   * @param block the block to compare and set
+   * @return true if the set operation succeeded, false if the block is already idle
+   */
+  bool ClearBlockBusyStatus(RawBlock *block) const {
+    uint32_t old_val = set_bit(block->insert_head_.load());
+    return block->insert_head_.compare_exchange_weak(old_val, clr_bit(old_val));
```

Comment:
Why case weak here. Will we lose a block forever in the face of a spurious failure?

Reply:
You are right, it should be strong